tests/postgresql_test.go

```diff
@@ -247,3 +249,76 @@ func TestPgCustomPrefix(t *testing.T) {
 	assert.NotEmpty(t, secret.Data["MY_PG_DATABASE_URI"])
 	assert.NotEmpty(t, secret.Data["MY_PG_CA_CERT"])
 }
+
+func getPgUpgradeVersionYaml(project, pgName, cloudName, version string) string {
+	return fmt.Sprintf(`
+  apiVersion: aiven.io/v1alpha1
+  kind: PostgreSQL
+  metadata:
+    name: %[2]s
+  spec:
+    authSecretRef:
+      name: aiven-token
+      key: token
+    project: %[1]s
+    cloudName: %[3]s
+    plan: startup-4
+    userConfig:
+      pg_version: "%[4]s"
+`, project, pgName, cloudName, version)
+}
+
+func TestPgUpgradeVersion(t *testing.T) {
+	t.Parallel()
+	defer recoverPanic(t)
+
+	// GIVEN
+	ctx := context.Background()
+	pgName := randName("secret-prefix")
+	yaml := getPgUpgradeVersionYaml(testProject, pgName, testPrimaryCloudName, "14")
+	s := NewSession(k8sClient, avnClient, testProject)
+
+	// Cleans test afterwards
+	defer s.Destroy()
+
+	// WHEN
+	// Applies given manifest
+	require.NoError(t, s.Apply(yaml))
+
+	// Waits kube objects
+	pg := new(v1alpha1.PostgreSQL)
+	require.NoError(t, s.GetRunning(pg, pgName))
+
+	// THEN
+	// Validates instance
+	pgAvn, err := avnClient.Services.Get(ctx, testProject, pgName)
+	require.NoError(t, err)
+	// Tests non-strict yaml. By sending string-integer we expect it's parsed as a string.
+	// Default version is 15, we get 14, as we set it.
+	assert.Equal(t, "14", pgAvn.UserConfig["pg_version"])
+	assert.Equal(t, anyPointer("14"), pg.Spec.UserConfig.PgVersion)
```

Comment:
The codgen client comes with version choices https://github.com/aiven/go-client-codegen/blob/main/handler/service/service.go#L1149
Technically you can create -2 and then upgrade to -1, and that will work when 14 is EOL.